File: sherlock-and-cost.py

t = int(input())

def func(a,b):
	if(abs(a-b)>abs(a-1)):
		return b
	else:
		return 1

for _ in range(t):
	n = input()
	n = int(n)

	a = []
	b = []
	b = list(map(int,input().split()))

	sum = 0
	if(b[0]>b[1]):
		a.append(b[0])
		a.append(1)
		sum+=b[0]-1
	else:
		a.append(1)
		a.append(b[1])
		sum+=b[1]-1

	for i in range(2,n):
		a.append(func(a[i-1],b[i]))
		sum+=abs(a[i]-a[i-1])

	print (sum)
	print (a)

File: sherlock-and-cost.cpp
#include<bits/stdc++.h>
using namespace std;
typedef long long int ll;

int main()
{
	freopen("input.txt","r",stdin);
	freopen("output.txt","w",stdout);
	ios::sync_with_stdio(0);

	ll t;
	cin>>t;

	while(t--)
	{
		ll n;
		cin>>n;

		ll arr[n+5],dp[n+5][2];
		for(ll i=0;i<n;i++)
		{
			cin>>arr[i];
			dp[i][0]=0;
			dp[i][1]=0;
		}

		for (ll i=1;i<n;i++)
		{
			dp[i][0] = max(dp[i-1][0],dp[i-1][1]+abs(arr[i-1]-1));
			dp[i][1] = max(dp[i-1][1]+abs(arr[i]-arr[i-1]),dp[i-1][0]+abs(arr[i]-1));
		}

		cout<<max(dp[n-1][0],dp[n-1][1])<<"\n";
	}
}